src/transports/http.rs

```diff
@@ -86,16 +91,15 @@ async fn execute_rpc<T: DeserializeOwned>(client: &Client, url: Url, request: &R
         .bytes()
         .await
         .map_err(|err| Error::Transport(format!("failed to read response bytes: {}", err)))?;
-    log::debug!(
-        "[id:{}] received response: {:?}",
-        id,
-        String::from_utf8_lossy(&response).as_ref()
-    );
+    let decoded_response = String::from_utf8_lossy(&response);
```

Comment:
I'd prefer if this decoding was only done if:
1. we print debug messages,
2. or we are in the error scenario.

Currently it will be converted to a string in a "happy path" as well, but the result is dropped.
I realize that in case of both (1) & (2) conditions being true, we might end up decoding twice, but my guts tell me this would be a better trade-off (happy to be proven otherwise with a benchmark though).

Reply:
I agree it's a better tradeoff, good point!